src/malli_ts/core.cljc

```diff
@@ -300,16 +310,55 @@
                    m [file]
                    (let [import (string/join "\n" (get file->import-literals file))
                          types (string/join "\n" (get file->type-literals file))]
-                     (str (if-not (string/blank? import) (str import "\n\n"))
+                     (str (if-not (string/blank? import) (str import "\n\n") nil)
                           types))))
                 {} files)]
     file-contents))
 
+(defn parse-matching-schemas
+  "Only applicable to qualified schema-types and not in defined in malli.core"
+  {:arglists '([options]
+               [pred options])}
+  ([pred {:keys [registry transform] :as options}]
+   (let [schemas (into []
+                       (comp (filter (fn [[k s]]
+                                       (and (qualified-keyword? k)
+                                            (not= "malli.core" (namespace k))
+                                            (pred k s))))
+                             (map (fn [[k _]] [k {}]))
+                             (map (or transform identity)))
+                       (mr/schemas (mr/composite-registry registry m/default-registry)))
+         parse-files-arg (persistent!
+                          (reduce
+                           (fn [acc [k opts]]
+                             (let [file-name (str (csk/->snake_case (namespace k)) ".d.ts")]
+                               (if-let [asdf (get acc file-name)]
+                                 (assoc! acc file-name (conj asdf [k opts]))
+                                 (assoc! acc file-name [[k opts]]))))
+                           (transient {}) schemas))]
+     (parse-files parse-files-arg options)))
+  ([options]
+   (parse-matching-schemas (constantly true) options))
+  ([]
+   (parse-matching-schemas (constantly true) {})))
+
+(defn parse-ns-schemas
+  ([ns-coll options]
+   (parse-matching-schemas
+    (let [ns-set (into #{} (map str) ns-coll)]
+      (fn [k _]
+        (let [k-ns (namespace k)]
+          (contains? ns-set k-ns))))
+    options))
+  ([ns-coll]
+   (parse-ns-schemas ns-coll {})))
+
 (defn external-type
   ([type-name type-path type-import-alias]
-   [any? {::external-type {:t-name type-name
-                           :t-path type-path
-                           :t-alias type-import-alias}}])
+   [any? {::external-type true
+          ::t-name type-name
+          ::t-path type-path
+          ::t-alias type-import-alias}])
```

Comment:
Should this be `::ts-name`  as it's specific to TypeScript?

Reply:
Or even `:ts/name`

Reply:
`t-name` stands for "type name", and it's explicit that is a typescript type name because the namespace is `malli-ts.core`

Reply:
I guess not 😄